controllers/userController.js

```diff
@@ -28,18 +28,133 @@ exports.getUser = function(req, res) {
 	});
 };
 
-exports.updateUser = function(req, res) {
-	var query = { '_id': req.user._id };
-	User.findOneAndUpdate(query, req.body.newData, { upsert:true }, function(err, user) {
-			if (err) {
-				return res.status(500).json({
-				 	message: err.message
-				})
-			}
-			res.send({
-				status: "ok",
-			});
-	});
+exports.updateUserSpritename = function(req, res) {
+  var query = { '_id': req.user._id };
+  if (!req.body.spritename) {
+    res.send({
+      status: "error",
+      error: "Please input new spritename for update"
+    });
+  }
+  var updateAttributes = { "spritename":req.body.spritename }
+  User.findOneAndUpdate(query, updateAttributes, { upsert:true }, function(err, user) {
+      if (err) {
+        return res.status(500).json({
+          message: err.message
+        })
+      }
+      res.send({
+        status: "ok",
+      });
+  });
+}
+
+//(strength, stamina, agility, health, healthLimit)
+exports.updateUserAttributes = function(req, res) {
+  var query = { '_id': req.user._id };
+  var updateAttributes = {}
+  if (req.body.strength) {
+    updateAttributes['strength'] = req.body.strength
+  }
+  if (req.body.stamina) {
+    updateAttributes['stamina'] = req.body.stamina
+  }
+  if (req.body.agility) {
+    updateAttributes['agility'] = req.body.agility
+  }
+  if (req.body.health) {
+    updateAttributes['health'] = req.body.health
+  }
+  if (req.body.healthLimit) {
+    updateAttributes['healthLimit'] = req.body.healthLimit
+  }
+  if (!updateAttributes.strength && !updateAttributes.stamina && !updateAttributes.agility && !updateAttributes.health && !updateAttributes.healthLimit) {
+    res.send({
```

Comment:
condition should be `||` (reject as long as one input is not correct)
Also, specify the condition explicitly (for example, must be >= 0)